PubNubUnity/Assets/PubNub/PlayModeTests/PlayModeTests.cs

```diff
@@ -848,4666 +923,5430 @@ public IEnumerator TestHereNowEmptyChannel()
+                        Assert.True("scv2" == mem.Space.Custom["spacecustomkey2"].ToString());
+                        if (withPAM)
+                        {
+                            Assert.True("mcv21" == mem.Custom["memberscustomkey21"].ToString());
+                            Assert.True("mcv22" == mem.Custom["memberscustomkey22"].ToString());
+                        }
+                        else
+                        {
+                            Assert.True("mcvup21" == mem.Custom["memberscustomkeyup21"].ToString());
+                            Assert.True("mcvup22" == mem.Custom["memberscustomkeyup22"].ToString());
+                        }
+
+                        bFound = true;
+                        break;
+                    }
+
+                }
 
-			tresult = false;
+                tresult = bFound;
+            });
 
-			pubnub.GetPresenceState().Channels(channelList).Async ((result, status) => {
-            //pubnub.SetPresenceState().Channels(new List<string> (){ch1}).State(state).Async ((result, status) => {    
-                if(status.Error){
-					Assert.Fail("GetPresenceState failed");
-                } else {
-                    if(result != null){
-						if(result.StateByChannels!= null){
-							bool found1= false, found2 = false;
-							foreach (KeyValuePair<string, object> key in result.StateByChannels){
-								Debug.Log("key.Key" + key.Key);
-								Dictionary<string, object> stateDict = key.Value as Dictionary<string, object>;
-								foreach (KeyValuePair<string, object> keyInStateDict in stateDict){
-									Debug.Log("keyInStateDict.Key" + keyInStateDict.Key);
-									if(keyInStateDict.Key.Equals(channel)){
-										bool stateMatch = pubnub.JsonLibrary.SerializeToJsonString(keyInStateDict.Value).Equals("{\"cg\":{\"k1\":\"v1\"}}");
-										Debug.Log("pubnub.JsonLibrary.SerializeToJsonString(keyInStateDict.Value)" + pubnub.JsonLibrary.SerializeToJsonString(keyInStateDict.Value));
-										Assert.IsFalse(stateMatch);
-										found1 = stateMatch;
-									}
-									if(keyInStateDict.Key.Equals(channel2)){
-										bool stateMatch = pubnub.JsonLibrary.SerializeToJsonString(keyInStateDict.Value).Equals("{\"cg\":{\"k1\":\"v1\"}}");
-										Debug.Log("pubnub.JsonLibrary.SerializeToJsonString(keyInStateDict.Value)" + pubnub.JsonLibrary.SerializeToJsonString(keyInStateDict.Value));
-										Assert.IsFalse(stateMatch);
-										found2 = stateMatch;
-									}
-								}
-							}
-							Assert.IsTrue(!found1);
-							Assert.IsTrue(!found2);
-							tresult = !found1 && !found2;
-						}
+            yield return new WaitForSeconds(PlayModeCommon.WaitTimeBetweenCalls4);
+            Assert.True(tresult, "GetMembershipsFilter didn't return");
+
+            // Sort feature test for GetMemberships
+            tresult = false;
+            pubnub.GetMemberships().UserID(userid).Include(inclMem).Sort(sort).Limit(1).Count(count).Async((result, status) =>
```

Comment:
@mohitpubnub please enhance the test to include that it's actually sorting the response.

Reply:
Yes, I should. So I am adding this to my TODO list